### DISCUSSION TITLE
TS011F_plug_1 switches off on its own

### DISCUSSION THREAD

**Discussion by andreamoretto** [Koenkk/zigbee2mqtt#11500] [Devices]
I have 3 of those plugs, and they spontaneously switches off on their own multiple times a day. I've noticed that, when this happens, the reported voltage is 0V, so my guess is that those devices misread the voltage at times.

Has anyone the same issue? Is it possible to update the firmware?

Thanks in advance,

Andrea

**Comment by woutercoppens** [comment]
I have the same issue :-(

**Comment by Koenkk** [comment]
Looks similar to https://github.com/Koenkk/zigbee2mqtt/issues/11648

**Reply by andreamoretto** [reply]
Yes, it definitely is! This TS011F plug version seems to be affected by some kind of bug in the voltage measurement. Sometimes the voltage drops to 0, and then the plug disconnects the load.

Is it possible to upgrade the plug's firmware?

**Reply by Koenkk** [reply]
@andreamoretto OTA is already enabled for some of these devices. If it shows no image error then follow https://www.zigbee2mqtt.io/advanced/more/tuya_xiaomi_ota_url.html

**Reply by giejay** [reply]
Just to understand, why would you need to go through this OTA url retrieval process if you can just update the devices using the Tuya app itself. You need the bridge anyway and the url's are really specific (with versioning). Or am I misreading this and would you need to go through this process only once?

**Reply by andreamoretto** [reply]
Unfortunately I don't own a tuya bridge yet, I run zigbee2mqtt using a E18-2G4U04B USB dongle (CC2531 based). Is it possible to do OTA without using the bridge or should I get one? Or, if I get the URL in another way (e.g. another user on this forum kindly provide it), is it possible to upgrade without the bridge?

**Reply by Koenkk** [reply]
> Or am I misreading this and would you need to go through this process only once?

Someone has to do it once, then other z2m users can use the same update. Unfortunately there are many different version of the TS011F plugs which means different OTAs for each.

**Reply by andreamoretto** [reply]
You are right, looks like there are many variant out there. Anyway, I will get a Tuya gateway and make the firmwares available to the community!

**Comment by woutercoppens** [comment]
I bought one of these plugs: https://nl.aliexpress.com/item/4001118546894.html?gatewayAdapt=glo2nld&spm=a2g0o.9042311.0.0.290f4c4dFw3ApM

The plug has version v1.0.10 and there are no upgrades available.
https://photos.app.goo.gl/HeCfz2gkduuQR1LJ9

**Reply by andreamoretto** [reply]
The problem is mitigated with firmware upgrade, as indicated in #11648.
